tests/integration/release/test_release.py

```diff
@@ -115,21 +145,152 @@ def test_release_row_noising(
+
+        prenoised_values = shared_prenoised.loc[to_compare_idx, column].values
+        noised_values = shared_noised.loc[to_compare_idx, column].values
+
+        different_check: npt.NDArray[np.bool_] = np.array(prenoised_values != noised_values)
+        noise_level = different_check.sum()
+
+        if includes_token_noising:
+            if noise_type == NOISE_TYPES.write_wrong_zipcode_digits.name:
+                token_probability: list[
+                    float
+                ] | int | float = config.get_zipcode_digit_probabilities(dataset_name, column)
+            else:
+                token_probability = config.get_token_probability(
+                    dataset_name, noise_type, column
+                )
+
+            # Get number of tokens per string to calculate expected proportion
+            tokens_per_string_getter: Callable[
+                ..., pd.Series[int] | int
+            ] = TOKENS_PER_STRING_MAPPER.get(noise_type, lambda x: x.astype(str).str.len())
+            tokens_per_string: pd.Series[int] | int = tokens_per_string_getter(
+                shared_prenoised.loc[to_compare_idx, column]
+            )
+            # Calculate probability no token is noised
+            if isinstance(token_probability, list):
+                # Calculate write wrong zipcode average digits probability any token is noise
+                avg_probability_any_token_noised = 1 - math.prod(
+                    [1 - p for p in token_probability]
+                )
+            else:
+                with check:
+                    assert isinstance(tokens_per_string, pd.Series)
+                avg_probability_any_token_noised = (
+                    1 - (1 - token_probability) ** tokens_per_string
+                ).mean()
+
+            # This is accumulating not_noised over all noise types
+            expected_noise = avg_probability_any_token_noised * expected_config_noise
+        else:
+            expected_noise = expected_config_noise
+
+        num_eligible = len(to_compare_idx)
+        # we sometimes copy the same column value from a household member so we only want
+        # to consider individuals who ended up with a different value as a result of this noising
+        can_silently_noise = noise_type == "copy_from_household_member" and (
+            column == "age" or column == "date_of_birth"
+        )
+        if can_silently_noise:
+            try:
+                num_sims_with_silent_noising = sum(
+                    shared_prenoised.loc[to_compare_idx, column].astype(float)
+                    == shared_prenoised.loc[to_compare_idx, f"copy_{column}"].astype(float)
+                )
+            except:
+                num_sims_with_silent_noising = sum(
+                    shared_prenoised.loc[to_compare_idx, column].astype(str)
+                    == shared_prenoised.loc[to_compare_idx, f"copy_{column}"].astype(str)
+                )
```

Comment:
We should dry this up. We should also catch the precise error that we are expecting, not just any error